lib/install/management/config.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package management
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/vmware/vic/pkg/trace"
+	"github.com/vmware/vic/pkg/vsphere/datastore"
+	"github.com/vmware/vic/pkg/vsphere/extraconfig"
+)
+
+var (
+	errSecretKeyNotFound = fmt.Errorf("unable to find %s in the .vmx file", extraconfig.GuestInfoSecretKey)
+	errNilDatastore      = fmt.Errorf("session's datastore is not set")
+	errInvalidTokens     = fmt.Errorf("error parsing .vmx file: unexpected token count in line")
+)
+
+// extractSecretFromFile reads and extracts the GuestInfoSecretKey value from the input.
+func extractSecretFromFile(rc io.ReadCloser) (string, error) {
+
+	scanner := bufio.NewScanner(rc)
+	for scanner.Scan() {
+		line := scanner.Text()
+
+		// The line is of the format: key = "value"
+		if strings.HasPrefix(line, extraconfig.GuestInfoSecretKey) {
+
+			fields := strings.Fields(line)
```

Comment:
Why still using strings.Fields rather than strings.SplitN?  In general, "=" is the delimiter, not whitespace.  While the secret value likely won't have spaces, it is still correct to split on "=", then Trim with the cutset `" ` (trims both double quotes + spaces)